snowpack/src/util.ts

```diff
@@ -36,9 +37,11 @@ export const SVELTE_VUE_REGEX = /(<script[^>]*>)(.*?)<\/script>/gims;
 
 export const URL_HAS_PROTOCOL_REGEX = /^(\w+:)?\/\//;
 
-const UTF8_FORMATS = ['.css', '.html', '.js', '.map', '.mjs', '.json', '.svg', '.txt', '.xml'];
-export function getEncodingType(ext: string): 'utf-8' | undefined {
-  return UTF8_FORMATS.includes(ext) ? 'utf-8' : undefined;
+/** Read file from disk; return a string if it’s a code file */
+export async function readFile(filepath: string): Promise<string | Buffer> {
```

Comment:
We replaced `getEncodingType` with a universal `readFile` function. The way `isbinaryfile` works, we actually need to read the file contents to determine if it’s a binary file or not, so it makes sense to combine them into one. The end result is this:

```js
const contents = await readFile(filepath);
if (typeof contents === 'string') {
 // This is a UTF-8 file (.js, .css, etc.)
} else {
 // This is a binary file (.jpg, .png, etc.)
}
```